arrow-libs/fx/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Fiber.kt

```diff
@@ -119,7 +119,14 @@ suspend fun <A> (suspend () -> A).forkConnected(ctx: CoroutineContext = Dispatch
  * }
  * ```
  */
-// TODO provide proper deprecation annotation
+@Deprecated(
+  "Use Deferred with KotlinX Coroutines Structured Concurrency",
+  ReplaceWith(
+    "val scope = CoroutineScope(ctx); scope.async(ctx) { f() }.also { scope.launch { interruptWhen(); it.cancelAndJoin() } }",
```

Comment:
this is a nice replacement, although I'm not sure I'd include the scope as that's something I believe people should create or obtain themselves depending on the type of app

Reply:
```suggestion
    "async(ctx) { f() }.also { launch { interruptWhen(); it.cancelAndJoin() } }",
```

Reply:
That howeer has diferren semantics. These operators followed a different approach as structured concurrency and therefore aren't really compatible.

Otherwise maybe:

`GlobalScope.async(ctx) { f() }.also { GlobalScope.launch { interruptWhen(); it.cancelAndJoin() } }` ?